File: LeetCode/Java/49_Group_Anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        
        for(String s: strs) {
            char[] charArray = s.toCharArray();
            Arrays.sort(charArray);
            String sortedString = new String(charArray);
            
            if(!map.containsKey(sortedString)) {
                map.put(sortedString, new LinkedList<String>());
            }
            map.get(sortedString).add(s);
        }
        return new LinkedList<>(map.values());
    }
}

File: LeetCode/C++/49. Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<char,int> inn;
        unordered_map< unordered_map<char,int> , vector<string>>hs;
        int n=strs.size();
        string temp;
        for(string s :strs)
        {
            for(int i=0;i<s.length();i++)
                inn[s[i]]++;
            hs[inn].push_back(s);
        }
        vector<vector<string>> result;
        for(auto i=hs.begin();i!=hs.end();i++)
            result.push_back(i->second);
        return result;
    }
};
